rmf_task_ros2/src/dispatcher_node/main.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2019 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <rclcpp/rclcpp.hpp>
+#include <rmf_task_ros2/dispatcher/Dispatcher.hpp>
+
+#include <rmf_task_msgs/srv/submit_task.hpp>
+#include <rmf_task_msgs/srv/cancel_task.hpp>
+#include <rmf_task_msgs/srv/get_task.hpp>
+
+//==============================================================================
+using SubmitTaskSrv = rmf_task_msgs::srv::SubmitTask;
+using CancelTaskSrv = rmf_task_msgs::srv::CancelTask;
+using GetTaskSrv = rmf_task_msgs::srv::GetTask;
+
+//==============================================================================
+class DispatcherNode : public rclcpp::Node
```

Comment:
I will caution against building the node this way, as you are essentially passing a `this` instance to be used and kept by the components directly owned by the node itself. The instances of action interface and auctioneer is currently holding onto a shared pointer to the same `DispatcherNode` that owns them.

Since we have multiple components that require an instance of a node, the dispatcher, the auctioneer, the action interface, etc. I will suggest composition over inheritance for `rclcpp::Node`, allow `DispatchNode` to hang onto a node as an implementation class member, and to pass that shared pointer of the node to it's various components, i.e. auctioneer and action interface.

Reply:
Thanks for the advice! Yah the api outbound api was not clearly defined. I have made those relevant changes in the newest commit, which this will have the node being constructed within the `Dispatcher.hpp` (Similar to `adapter.hpp`).